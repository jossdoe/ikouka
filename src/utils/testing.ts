// This is a proxy-file to allow for global wrappers and modifications
// to the functionality of React Testing Library.
// https://testing-library.com/docs/react-testing-library/setup#custom-render
export * from '@testing-library/react'
