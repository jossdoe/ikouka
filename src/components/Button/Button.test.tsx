import React from 'react'
import { render, screen } from '../../utils/testing'
import userEvent from '@testing-library/user-event'
import { Button, ButtonProps } from './index'

const LABEL_TEXT = 'Click Me'

const VARIANTS: ButtonProps['variant'][] = [
  'primary',
  'secondary',
  'text',
  undefined
]

const SIZES: ButtonProps['size'][] = ['sm', 'md', 'lg', undefined]

VARIANTS.forEach((variant) => {
  SIZES.forEach((size) => {
    test(`Button (onClick) – ${variant}, ${size}`, async () => {
      const onClick = jest.fn()

      render(
        <Button variant={variant} size={size} onClick={onClick}>
          {LABEL_TEXT}
        </Button>
      )

      expect(screen.getByRole('button')).toHaveTextContent(LABEL_TEXT)

      expect(onClick.mock.calls).toHaveLength(0)
      await userEvent.click(screen.getByText(LABEL_TEXT))
      expect(onClick.mock.calls).toHaveLength(1)
    })
  })
})

VARIANTS.forEach((variant) => {
  SIZES.forEach((size) => {
    test(`Button (href) – ${variant}, ${size}`, async () => {
      render(
        <Button variant={variant} size={size} href="#">
          {LABEL_TEXT}
        </Button>
      )

      expect(screen.getByRole('link')).toHaveTextContent(LABEL_TEXT)
      expect(screen.getByRole('link')).toHaveAccessibleName(LABEL_TEXT)
      expect(screen.getByRole('link')).toHaveAttribute('href', '#')
      expect(screen.getByRole('link')).toHaveAttribute('target', '_blank')
      expect(screen.getByRole('link')).toHaveAttribute(
        'rel',
        'noopener noreferrer'
      )
    })
  })
})
