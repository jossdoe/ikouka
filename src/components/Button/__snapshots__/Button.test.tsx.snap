// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button (href) – primary, lg 1`] = `
<DocumentFragment>
  <a
    class="px-6 py-2.5 text-lg rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – primary, md 1`] = `
<DocumentFragment>
  <a
    class="px-4 py-2 text-base rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – primary, sm 1`] = `
<DocumentFragment>
  <a
    class="px-3 py-1.5 text-sm rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – primary, undefined 1`] = `
<DocumentFragment>
  <a
    class="px-4 py-2 text-base rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – secondary, lg 1`] = `
<DocumentFragment>
  <a
    class="px-6 py-2.5 text-lg rounded-lg bg-gray-100 text-gray-600 hover:bg-gray-200 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – secondary, md 1`] = `
<DocumentFragment>
  <a
    class="px-4 py-2 text-base rounded-lg bg-gray-100 text-gray-600 hover:bg-gray-200 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – secondary, sm 1`] = `
<DocumentFragment>
  <a
    class="px-3 py-1.5 text-sm rounded-lg bg-gray-100 text-gray-600 hover:bg-gray-200 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – secondary, undefined 1`] = `
<DocumentFragment>
  <a
    class="px-4 py-2 text-base rounded-lg bg-gray-100 text-gray-600 hover:bg-gray-200 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – text, lg 1`] = `
<DocumentFragment>
  <a
    class="px-6 py-2.5 text-lg rounded-lg bg-transparent text-gray-600 hover:bg-gray-100 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – text, md 1`] = `
<DocumentFragment>
  <a
    class="px-4 py-2 text-base rounded-lg bg-transparent text-gray-600 hover:bg-gray-100 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – text, sm 1`] = `
<DocumentFragment>
  <a
    class="px-3 py-1.5 text-sm rounded-lg bg-transparent text-gray-600 hover:bg-gray-100 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – text, undefined 1`] = `
<DocumentFragment>
  <a
    class="px-4 py-2 text-base rounded-lg bg-transparent text-gray-600 hover:bg-gray-100 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – undefined, lg 1`] = `
<DocumentFragment>
  <a
    class="px-6 py-2.5 text-lg rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – undefined, md 1`] = `
<DocumentFragment>
  <a
    class="px-4 py-2 text-base rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – undefined, sm 1`] = `
<DocumentFragment>
  <a
    class="px-3 py-1.5 text-sm rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (href) – undefined, undefined 1`] = `
<DocumentFragment>
  <a
    class="px-4 py-2 text-base rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    Click Me
  </a>
</DocumentFragment>
`;

exports[`Button (onClick) – primary, lg 1`] = `
<DocumentFragment>
  <button
    class="px-6 py-2.5 text-lg rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – primary, md 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 text-base rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – primary, sm 1`] = `
<DocumentFragment>
  <button
    class="px-3 py-1.5 text-sm rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – primary, undefined 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 text-base rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – secondary, lg 1`] = `
<DocumentFragment>
  <button
    class="px-6 py-2.5 text-lg rounded-lg bg-gray-100 text-gray-600 hover:bg-gray-200 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – secondary, md 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 text-base rounded-lg bg-gray-100 text-gray-600 hover:bg-gray-200 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – secondary, sm 1`] = `
<DocumentFragment>
  <button
    class="px-3 py-1.5 text-sm rounded-lg bg-gray-100 text-gray-600 hover:bg-gray-200 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – secondary, undefined 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 text-base rounded-lg bg-gray-100 text-gray-600 hover:bg-gray-200 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – text, lg 1`] = `
<DocumentFragment>
  <button
    class="px-6 py-2.5 text-lg rounded-lg bg-transparent text-gray-600 hover:bg-gray-100 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – text, md 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 text-base rounded-lg bg-transparent text-gray-600 hover:bg-gray-100 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – text, sm 1`] = `
<DocumentFragment>
  <button
    class="px-3 py-1.5 text-sm rounded-lg bg-transparent text-gray-600 hover:bg-gray-100 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – text, undefined 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 text-base rounded-lg bg-transparent text-gray-600 hover:bg-gray-100 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – undefined, lg 1`] = `
<DocumentFragment>
  <button
    class="px-6 py-2.5 text-lg rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – undefined, md 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 text-base rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – undefined, sm 1`] = `
<DocumentFragment>
  <button
    class="px-3 py-1.5 text-sm rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button (onClick) – undefined, undefined 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 text-base rounded-lg bg-blue-100 text-blue-600 hover:bg-blue-200 "
  >
    Click Me
  </button>
</DocumentFragment>
`;
